azure-proto-core/Resources/Plan.cs

```diff
@@ -17,12 +16,43 @@ public class Plan : IEquatable<Plan>, IComparable<Plan>
 
         public int CompareTo(Plan other)
         {
-            throw new NotImplementedException();
+            if (other == null) return 1;
+            if (this.Name.CompareTo(other.Name) == 0)
+            {
```

Comment:
This nesting is pretty intense and hard to read. This can be accomplished without this nesting. 

if not x(){
Return xxx
}
if not y(){
return yyy
}

last line would then be the nested line:   return this.Version.CompareTo(other.Version);

Reply:
Good point, will update